src/main/java/com/iota/iri/service/snapshot/impl/LocalSnapshotManagerImpl.java

```diff
@@ -136,4 +146,46 @@ private void monitorThread(LatestMilestoneTracker latestMilestoneTracker) {
             ThreadUtils.sleep(LOCAL_SNAPSHOT_RESCAN_INTERVAL);
         }
     }
+    
+    /**
+     * A snapshot is taken in an interval. 
+     * This interval changes based on the state of the node.
+     * 
+     * @param inSync if this node is in sync
+     * @return the current interval in which we take local snapshots
+     */
+    private int getSnapshotInterval(boolean inSync) {
+        return inSync
+                ? config.getLocalSnapshotsIntervalSynced()
+                : config.getLocalSnapshotsIntervalUnsynced();
+    }
+    
+    /**
+     * A node is defined in sync when the latest snapshot milestone index and the latest milestone index are equal.
+     * In order to prevent a bounce between in and out of sync, a buffer is added when a node became in sync.
+     * 
+     * This will always return false if we are not done scanning milestone candidates during initialization.
+     * 
+     * @param latestMilestoneTracker tracker we use to determine milestones
+     * @return <code>true</code> if we are in sync, otherwise <code>false</code>
+     */
+    private boolean isInSync(LatestMilestoneTracker latestMilestoneTracker) {
+        if (!latestMilestoneTracker.isInitialScanComplete()) {
+            return false;
+        }
+        
+        int latestIndex = latestMilestoneTracker.getLatestMilestoneIndex();
+        int latestSnapshot = snapshotProvider.getLatestSnapshot().getIndex();
+        
+        // If we are out of sync, only a full sync will get us in
+        if (!isInSync && latestIndex == latestSnapshot) {
+            isInSync = true;
+        
+        // When we are in sync, only dropping below the buffer gets us out of sync 
+        } else if (latestSnapshot < latestIndex - LOCAL_SNAPSHOT_SYNC_BUFFER) {
+            isInSync = false;
+        }
```

Comment:
This could be made simpler:
```
if (latestSnapshot < latestIndex - LOCAL_SNAPSHOT_SYNC_BUFFER) {
   return false;
}
 else {
    return true;
}
```

Reply:
You cant right? Because then instead of boucing at `latestSnapshot` and `latestIndex`, it will bounce at `latestIndex - BUFFEr` and `latestsnapshot`